src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -36,25 +35,15 @@ public static VisualStudioInstance GetInstance(string workingDirectory)
                 return null;
             }
 
-            string versionContent = File.ReadAllText(versionPath);
-            Match versionMatch = VersionRegex.Match(versionContent);
-
-            if (!versionMatch.Success)
-            {
-                return null;
-            }
-
-            if (!int.TryParse(versionMatch.Groups[1].Value, out int major) ||
-                !int.TryParse(versionMatch.Groups[2].Value, out int minor) ||
-                !int.TryParse(versionMatch.Groups[3].Value, out int patch))
+            if (!Version.TryParse(File.ReadAllText(versionPath), out Version version) || version > Environment.Version)
```

Comment:
`Environment.Version` is a strange beast. It's **always** `4.0.30319.42000` for .NET 4.6 and higher and for .NET Core 1.0-3.1. Starting with .NET 5.0, it returns a value more like you'd probably expect: https://github.com/dotnet/coreclr/pull/22664.

FORTUNATELY, since this PR is mostly about blocking out 5.0 versions from 3.1 hosts (since people mostly didn't complain about 3.1 versions for 2.1 hosts) this simple call should still work. And keep working into the future for 6.0/7.0 on 5.0 and so on.